toto/runlib.py

```diff
@@ -100,6 +100,11 @@ def record_artifacts_as_dict(artifacts):
     return artifacts_dict
 
   for artifact in artifacts:
+
+    if not os.path.exists(artifact):
+      log.warning("path: {} does not exist, skipping..".format(artifact))
+      continue 
+
```

Comment:
Hmm as an addition? So this would be like the following:
```
if is file:
     record
elif isdict:
     record using walk
elif:
     record 
     log.warn("Attempted to record unknown filetype {}".format(artifact))
```
Or you're thinking of dropping the os.path.exists() altogether?